### ISSUE TITLE
Security group rule description not validated

### ISSUE THREAD

**Issue by j-boivie** [aws-cloudformation/cfn-python-lint#1471]
_cfn-lint version: cfn-lint 0.29.5_

I came across a small problem similar to the issue described in #173. Security group _rule descriptions_ have some constraints on characters/length allowed ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-rule-1.html#cfn-ec2-security-group-rule-description)) and it seems like these are not being checked by cfn-lint.

The following snippet will be accepted by cfn-lint but cause an issue during deployment:

```yaml
LbSecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: Load Balancer SG
      VpcId: !Ref VPC
      SecurityGroupIngress:
        - Description: -> doesn't work
          CidrIp: 0.0.0.0/0
          FromPort: 80
          ToPort: 80
```
Which would result in this:
```Invalid rule description. Valid descriptions are strings less than 256 characters from the following set: a-zA-Z0-9. _-:/()#,@[]+=&;{}!$*```

Unsure if this also applies when using Ingress rule resource `AWS::EC2::SecurityGroupIngress` as there is no similar constraint in the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-ingress.html#cfn-ec2-security-group-ingress-description).

Thanks!

**Comment by PatMyron** [comment]
As [described here](https://github.com/aws-cloudformation/cfn-python-lint/issues/903#issuecomment-595356853), this would need something like:

https://github.com/aws-cloudformation/cfn-python-lint/blob/a46773c752247c51effef415bd3462eaec10ab0b/src/cfnlint/data/ExtendedSpecs/all/03_value_types.json#L33-L36

https://github.com/aws-cloudformation/cfn-python-lint/blob/a46773c752247c51effef415bd3462eaec10ab0b/src/cfnlint/data/ExtendedSpecs/all/04_property_values.json#L330-L336

and [the rest of the necessary changes](https://github.com/aws-cloudformation/cfn-python-lint/tree/master/src/cfnlint/data/CloudSpecs) should be auto-generatable by running:

```shell
cfn-python-lint $ pip3 install -e . --user
cfn-python-lint $ cfn-lint --update-specs
```

---

There should be no functional difference, but [`src/cfnlint/data/ExtendedSpecs/all/03_value_types`](https://github.com/aws-cloudformation/cfn-python-lint/tree/master/src/cfnlint/data/ExtendedSpecs/all/03_value_types) and [`src/cfnlint/data/ExtendedSpecs/all/04_property_values`](https://github.com/aws-cloudformation/cfn-python-lint/tree/master/src/cfnlint/data/ExtendedSpecs/all/04_property_values) are more organized than [`src/cfnlint/data/ExtendedSpecs/all/03_value_types.json`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/all/03_value_types.json) and [`src/cfnlint/data/ExtendedSpecs/all/04_property_values.json`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/ExtendedSpecs/all/04_property_values.json), so they should be preferred locations for new constraints
